src/main/java/dk/cs/aau/huppaal/code_analysis/CodeAnalysis.java

```diff
@@ -1,15 +1,22 @@
 package dk.cs.aau.huppaal.code_analysis;
 
 import dk.cs.aau.huppaal.abstractions.Component;
+import dk.cs.aau.huppaal.logging.Log;
+import dk.cs.aau.huppaal.logging.LogLevel;
 import javafx.beans.property.SimpleStringProperty;
 import javafx.beans.property.StringProperty;
 import javafx.collections.FXCollections;
 import javafx.collections.ObservableList;
 
 import java.util.*;
 
+/**
+ * Logging system for analysis conclusions to be presented.
+ * e.g. if a location has no incoming edges, that indicates an implementation error
+ * @deprecated Use the new logging framework ({@link Log}) for this
+ * */
+@Deprecated
 public class CodeAnalysis {
```

Comment:
Are all the tasked previously performed by this class implemented for the new logging system?

Reply:
I also think using `logging` as the terminology is better for this feature, as proposed by the PR :+1:

Reply:
I don't have a good grasp of usages of the `CodeAnalysis` class / architecture and it would be a bit too much work to refactor all usages to use the new logging system. So I've just attached some `Log.addX` to the areas where messages was "logged" before.

Reply:
Looking forward to that :+1: